node_shell.py

```diff
@@ -387,10 +851,271 @@ def do_exit(self, arg):  # pylint: disable=unused-argument
+                self.poutput(AiiDANodeShell.do_verdi.__doc__)
+            # Passing -p is not allowed, we can only use the current profile
+            if passed_args[0] in ('-p', '--profile'):
+                self.poutput(
+                    red('Error: ') +
+                    'Manual profile selection is not allowed in the node shell.'
+                )
+                self.poutput(
+                    'To switch profile, relaunch the node shell with verdi -p <profile} run.'
+                )
+                self.poutput('Your current profile is: {}'.format(
+                    green(self.current_profile)))
+                return
+
+        verdi_args.extend(passed_args)
+        try:
+            with self.verdi_isolate():
+                verdi.main(args=verdi_args, prog_name='verdi')
+        except SystemExit as exception:
+            # SystemExit means the command-line tool finished as intented
+            # No action needed
+            pass
+        except Exception as exception:
+            # Catch all python exceptions raised during the verdi command execution
+            self.poutput(
+                'Verdi Command raised an exception {}'.format(exception))
+            print_exc(file=self.stdout)
+
+    group_parser = cmd2.Cmd2ArgumentParser()
+
+    group_parser.add_argument('--user-email', '-u', help='Filter by user')
+    group_parser.add_argument('--all-users',
+                              help='Flag if include all users',
+                              action='store_true')
+    group_parser.add_argument('--all-types',
+                              '-a',
+                              help='Flag if include all types',
+                              action='store_true')
+    group_parser.add_argument('--group-type',
+                              '-t',
+                              default=GroupTypeString.USER.value,
+                              help='Filter by type')
+    group_parser.add_argument('--with-description',
+                              '-d',
+                              help='Show also the decription',
+                              action='store_true')
+    group_parser.add_argument('--startswith',
+                              '-s',
+                              help='Filter by the initial string')
+    group_parser.add_argument('--endswith',
+                              '-e',
+                              help='Filter by ending string')
+    group_parser.add_argument(
+        '--contains',
+        '-c',
+        help='Filter by checking if the group name contains the string')
+    group_parser.add_argument(
+        '--past-days',
+        '-p',
+        type=int,
+        help='Filter by only including those created in the past N days')
+    group_parser.add_argument(
+        '--count',
+        '-C',
+        action='store_true',
+        help='Show also the number of nodes in the group')
+
+    @cmd2.with_argparser(group_parser)
+    def do_group_list(self, args):
```

Comment:
Same comment here - let's just remove this wrapper and people can now do `verdi group list`